service/src/main/java/org/jboss/sbomer/service/feature/sbom/pyxis/PyxisValidatingClient.java

```diff
@@ -0,0 +1,100 @@
+
+import java.util.List;
+import java.util.Set;
+
+import jakarta.validation.ConstraintViolationException;
+
+import org.eclipse.microprofile.faulttolerance.Retry;
+
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import org.jboss.sbomer.service.feature.sbom.pyxis.dto.PyxisRepository;
+import org.jboss.sbomer.service.feature.sbom.pyxis.dto.PyxisRepositoryDetails;
+import org.jboss.sbomer.service.rest.faulttolerance.RetryLogger;
+
+import io.smallrye.faulttolerance.api.BeforeRetry;
+import io.smallrye.faulttolerance.api.FibonacciBackoff;
+
+import jakarta.validation.ConstraintViolation;
+
+import jakarta.validation.Validator;
+
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.QueryParam;
+import jakarta.enterprise.context.ApplicationScoped;
+
+import jakarta.inject.Inject;
+
+import static org.jboss.sbomer.service.rest.faulttolerance.Constants.PYXIS_UNPUBLISHED_MAX_RETRIES;
+import static org.jboss.sbomer.service.rest.faulttolerance.Constants.PYXIS_UNPUBLISHED_INITIAL_DELAY;
+import static org.jboss.sbomer.service.rest.faulttolerance.Constants.PYXIS_UNPUBLISHED_MAX_DELAY;
+import static org.jboss.sbomer.service.rest.faulttolerance.Constants.PYXIS_UNPUBLISHED_MAX_DURATION;
+
+@ApplicationScoped
+public class PyxisValidatingClient {
+
+    @RestClient
+    PyxisService p;
+
+    @Inject
+    Validator validator;
+
+    @Retry(
+            maxRetries = PYXIS_UNPUBLISHED_MAX_RETRIES,
+            delay = PYXIS_UNPUBLISHED_INITIAL_DELAY,
+            maxDuration = PYXIS_UNPUBLISHED_MAX_DURATION,
+            retryOn = ConstraintViolationException.class)
+    @BeforeRetry(RetryLogger.class)
+    @FibonacciBackoff(maxDelay = PYXIS_UNPUBLISHED_MAX_DELAY)
+    public PyxisRepositoryDetails getRepositoriesDetails(
+            @PathParam("nvr") String nvr,
+            @QueryParam("include") List<String> includes) {
+        PyxisRepositoryDetails prd = p.getRepositoriesDetails(nvr, includes);
+        Set<ConstraintViolation<PyxisRepositoryDetails>> violations = validator.validate(prd);
+        if (violations.isEmpty()) {
+            return prd;
+        } else {
+            throw new ConstraintViolationException("Pyxis Repository constraint failure", violations);
+        }
+    }
+
+    @Retry(
+            maxRetries = PYXIS_UNPUBLISHED_MAX_RETRIES,
+            delay = PYXIS_UNPUBLISHED_INITIAL_DELAY,
+            maxDuration = PYXIS_UNPUBLISHED_MAX_DURATION,
+            retryOn = ConstraintViolationException.class)
+    @BeforeRetry(RetryLogger.class)
+    @FibonacciBackoff(maxDelay = PYXIS_UNPUBLISHED_MAX_DELAY)
+    public PyxisRepository getRepository(
+            @PathParam("registry") String registry,
+            @PathParam("repository") String repository,
+            @QueryParam("include") List<String> includes) {
+        PyxisRepository pr = p.getRepository(registry, repository, includes);
+        Set<ConstraintViolation<PyxisRepository>> violations = validator.validate(pr);
```

Comment:
While I can see a `jakarta.validation.constraints.NotEmpty` contraint on the `RepositoryDetail`, I don't see any constraint on the `PyxisRepository`, what condition is supposed to be throwing the exception?

Reply:
BTW @jonathanchristison , I just read a comment inside `ReleaseStandardAdvisoryEventsListener` which says that `getRepository` is not used at the moment.... so you can probably nuke it directly :-)